src/braket/devices/local_simulator.py

```diff
@@ -326,3 +341,19 @@ def _(
             )
         results = simulator.run(program, shots, *args, **kwargs)
         return AnalogHamiltonianSimulationQuantumTaskResult.from_object(results)
+
+    def _validate_noise_model_support(self) -> None:
```

Comment:
String parsing for validation is always tricky. How much can we guarantee that the names will match?

Reply:
It will work as long as the class name of the noise instruction matches the pragma name defined in `_to_openqasm()`.

Reply:
I agree that string parsing is tricky. But there is also no other attribute to tell whether a device support noise or not. Alternatives that I have considered are
1. Do not validate. The noise model can be added to SV1 or QPU (which do not support noise). BDK does not throw error. The circuit with noise model applied is only validated at the service. 
2. Add a `openqasm_pragma_name` attribute to connect the name used in `_to_openqasm()` and the name in `supportedPragma`. This seems to make the Noise class clunkier.

Reply:
@speller26 Do you have any thoughts on validation? Adding `openqasm_pragma_name` may remove string parsing. But it still need to match the pragma name in service.

Reply:
This is only for local simulator, right? So noise model can't be added to SV1 or QPU?

Reply:
Unfortunately, getting any information about supported operations relies on some form of string matching; I do like the idea of adding a [translation](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/src/braket/circuits/translations.py) for pragma names.

As for device support, SV1 and QPUs don't apply, but noise models do make sense for DM1.

Reply:
> As for device support, SV1 and QPUs don't apply, but noise models do make sense for DM1.

I meant this class only applies to local sim, so how we validate doesn't need to be tied to functionality of adding a noise model to things that don't support noise, or am I missing something?

Reply:
I have updated the PR to 
1. Use the circuit.translation to match pragma names
2. Add support to AwsDevice